ui/src/map/plan-download/plan-download-area-section.tsx

```diff
@@ -0,0 +1,256 @@
+import * as React from 'react';
+import { useTranslation } from 'react-i18next';
+import {
+    LocationTrackItemValue,
+    SearchDropdown,
+    SearchItemValue,
+    SearchType,
+    TrackNumberItemValue,
+} from 'tool-bar/search-dropdown';
+import { LayoutContext } from 'common/common-model';
+import { LayoutLocationTrack, LayoutTrackNumber } from 'track-layout/track-layout-model';
+import { AreaSelection, PlanDownloadState } from 'map/plan-download/plan-download-store';
+import {
+    filterErrors,
+    filterWarnings,
+    getVisibleErrorsByProp,
+    getVisibleIssuesAndParamsByProp,
+    hasErrors,
+    PropEdit,
+} from 'utils/validation-utils';
+import { createClassName } from 'vayla-design-lib/utils';
+import styles from 'map/plan-download/plan-download-popup.scss';
+import { Radio } from 'vayla-design-lib/radio/radio';
+import { FieldLayout } from 'vayla-design-lib/field-layout/field-layout';
+import { TextField } from 'vayla-design-lib/text-field/text-field';
+import { Spinner } from 'vayla-design-lib/spinner/spinner';
+
+type AreaSelectionType = 'TRACK_METERS' | 'MAINTENANCE_AREA' | 'MAP_AREA';
+const ASSET_SEARCH_TYPES = [SearchType.TRACK_NUMBER, SearchType.LOCATION_TRACK];
+
+export const PlanDownloadAreaSection: React.FC<{
+    layoutContext: LayoutContext;
+    trackNumber: LayoutTrackNumber | undefined;
+    locationTrack: LayoutLocationTrack | undefined;
+    state: PlanDownloadState;
+    onUpdateProp: <TKey extends keyof AreaSelection>(
+        propEdit: PropEdit<AreaSelection, TKey>,
+    ) => void;
+    onCommitField: (field: keyof AreaSelection) => void;
+    loading: boolean;
+}> = ({
+    layoutContext,
+    locationTrack,
+    trackNumber,
+    state,
+    onUpdateProp,
+    onCommitField,
+    loading,
+}) => {
+    const { t } = useTranslation();
+    const [areaSelectionType, setAreaSelectionType] =
+        React.useState<AreaSelectionType>('TRACK_METERS');
+
+    const updateProp = <TKey extends keyof AreaSelection>(
+        key: TKey,
+        value: AreaSelection[TKey],
+    ) => {
+        onUpdateProp({
+            key: key,
+            value: value,
+            editingExistingValue: false,
+        });
+    };
+
+    const onItemSelected = (item: SearchItemValue) => {
+        if (item) {
+            if (item.type === 'trackNumberSearchItem') {
+                updateProp('trackNumber', item.trackNumber.id);
+                updateProp('locationTrack', undefined);
+            } else if (item.type === 'locationTrackSearchItem') {
+                updateProp('locationTrack', item.locationTrack.id);
+                updateProp('trackNumber', undefined);
+            }
+        }
+    };
+    const hasEndBeforeStartError = getVisibleErrorsByProp(
+        state.committedFields,
+        state.validationIssues,
+        'endTrackMeter',
+    ).includes('end-before-start');
```

Comment:
Tämä on ehkä hieman rumaa koodia, mutta halusin että siinä tilanteessa että alkuosoite on loppuosoitteen jälkeen molemmat kentät muuttuvat punaisiksi virheen merkiksi, mutta että ainoastaan toisessa näytetään virheteksti

Reply:
Joo. Note to self: Tämän voisi constailla niin, että myös virheen luonnissa käytetään samaa constia